tracex_project/tracex/tests/test_utils.py

```diff
@@ -0,0 +1,162 @@
+        self.assertIn("End Timestamp", df_renamed.columns)
+        self.assertIn("Duration", df_renamed.columns)
+        self.assertIn("Location", df_renamed.columns)
+
+    def test_create_html_table_from_df(self):
+        """Tests if the html table is correctly created from the dataframe."""
+        data = {
+            "case:concept:name": [1, 1],
+            "activity": ["fell ill", "went to the doctor"],
+            "time:timestamp": ["2021-06-05", "2021-06-06"],
+        }
+        df = pd.DataFrame(data)
+        html_table = Conversion.create_html_table_from_df(df)
+
+        self.assertIn("<table", html_table)
+
+    def test_create_dfg_from_df(self):
+        """Tests if the dfg image is correctly created from the dataframe."""
+        data = {
+            "case:concept:name": [1, 1],
+            "activity": ["fell ill", "went to the doctor"],
+            "time:timestamp": ["20210605T0000", "20210606T0000"],
+            "event_type": ["Symptom Onset", "Doctor Visit"],
+        }
+        df = pd.DataFrame(data)
+        df["time:timestamp"] = pd.to_datetime(
+            df["time:timestamp"], format="%Y%m%dT%H%M", errors="coerce"
+        )
+        dfg_img = Conversion.create_dfg_from_df(df, "event_type")
+
+        self.assertIsNot(None, dfg_img)
+
+    def test_dataframe_to_xes(self):
+        """Tests if the dataframe is converted to xes under the right filepath."""
+        data = {
+            "case:concept:name": [1, 1],
+            "activity": ["fell ill", "went to the doctor"],
+            "time:timestamp": ["2021-06-05", "2021-06-06"],
+            "event_type": ["Symptom Onset", "Doctor Visit"],
+        }
+        df = pd.DataFrame(data)
+        file_path = Conversion.dataframe_to_xes(df, "test", "event_type")
+        with open(file_path, "r") as file:
+            file_content = file.read()
+
+        self.assertIn("test.xes", file_path)
+        self.assertIn('<?xml version="1.0" encoding="utf-8" ?>', file_content)
+        self.assertIn("<log", file_content)
+
+
+class DataframeUtilitiesTests(TestCase):
+    """Test cases for the dataframe utilities inside the utils module."""
+
+    fixtures = ["tracex_project/tracex/fixtures/dataframe_fixtures.json"]
+
+    def test_get_events_df(self):
+        """Tests if get_events_df returns all events in a dataframe correctly."""
+        events_df = DataFrameUtilities.get_events_df()
+
+        self.assertIsInstance(events_df, pd.DataFrame)
+        self.assertIn("case:concept:name", events_df.columns)
+        self.assertIn("activity", events_df.columns)
+        self.assertIn("time:timestamp", events_df.columns)
+        self.assertIn("time:end_timestamp", events_df.columns)
+        self.assertIn("event_type", events_df.columns)
+        self.assertIn("attribute_location", events_df.columns)
+
+    def test_get_events_df_with_query(self):
+        """Test if get_events_df returns queried events in a dataframe correctly."""
```

Comment:
"""Test if get_events_df returns queried events in a dataframe correctly."""
```suggestion
        """Tests if get_events_df returns queried events in a dataframe correctly."""
```